smart_contracts/contract/src/contract_api/runtime.rs

```diff
@@ -274,6 +274,31 @@ pub fn remove_key(name: &str) {
     unsafe { ext_ffi::casper_remove_key(name_ptr, name_size) }
 }
 
+/// Returns the set of [`AccountHash`] from the calling account's context `authorization_keys`.
+pub fn list_authorization_keys() -> BTreeSet<AccountHash> {
```

Comment:
I know why it is the case, but on reflection it seems confusing to me that the helper method is called `list_authorization_**keys**` but it returns a set of `AccountHash`.

would it make sense to turn the AccountHash's into Key::Account to align the naming w the type?

Reply:
OK, this would be a BIG change but I think we should be returning `BTreeSet<PublicKey>` here. Mostly b/c we can derive an `AccountHash` from `PublicKey` but not the other way around. It technically is possible b/c `deploy` [uses `PublicKey` type](https://github.com/casper-network/casper-node/blob/dev/node/src/types/deploy.rs#L1302) that gets mapped to an `AccountHash` when creating a `DeployItem`. The change would be mostly internal and we can keep the interfaces we have so far intact and add new methods that return `PublicKey`s instead. I think contract authors would be more interested in the keys rather than hashes.

Reply:
> I know why it is the case, but on reflection it seems confusing to me that the helper method is called `list_authorization_**keys**` but it returns a set of `AccountHash`.
> 
> would it make sense to turn the AccountHash's into Key::Account to align the naming w the type?

So this approach would require extra work to unwrap `AccountHash` on the Wasm side. Before making a decision, it's worth pointing out that we also have existing APIs with `_keys` suffix: `add_associated_key,` `remove_associated_key,` `update_associated_key` all operate on `AccountHash`es.